### DISCUSSION TITLE
User group assignments updated only after logout-login

### DISCUSSION THREAD

**Discussion by JussiKivimaki** [requarks/wiki#3507] [Error / Bug Report]
**Describe the bug**
I've created couple groups and different permissions for them (page rules too). Tested that the permissions work as expected. However, when I assign a group to a user, the group behavior/permissions doesn't update unless the user logs out and logs in again. After login everything works like it should. 

Without logging out, when the user opens the user proifle page, the updated user group is seen there correctly but the permissions doesn't behave like they should.

**To Reproduce**
1. Create a new group that has different permissions and page rules than say a default one (mine had permissions to write and edit pages, default was able only to view pages)
2. Log in user that has that default group, keep logged in
3. Assign the user to that user group. I tested both having both the default group and new group at the same time and only the new user group assigned.
4. Try to do something with the new permissions that should be available (i.e. create pages)
5. Not working? Log the user out and log in again. Works. 

**Expected behavior**
When assigning user to a group, I'd expect changes take place immediately and not require users to logout and login.

**Host Info (please complete the following information):**
 - OS: Linux
 - Wiki.js version: 2.5.159
 - Database engine: postgres 13.0

**Additional context**
Tested on above host with test user authenticated with slack. Tested also on local windows/2.5.159 instal with the local auth strategy too.

**Comment by MKraust** [comment]
Same here. It because permissions are written in JWT. Looks like token isn't invalidated on permissions change.

**Reply by NGPixel** [reply]
You can lower the JWT short expiration date in Admin > Security.
